module/tools/SystemConfiguration/src/SystemConfiguration.cpp

```diff
@@ -0,0 +1,212 @@
+                    fs::path out_file     = "/" / relative;
+
+                    // If we have a device specific override use that
+                    if (fs::is_regular_file(device_file)) {
+                        in_file = device_file;
+                        log<NUClear::TRACE>(fmt::format("Using {} config for {}", hostname, out_file.string()));
+                    }
+                    else {
+                        in_file = default_file;
+                        log<NUClear::TRACE>(fmt::format("Using default config for {}", out_file.string()));
+                    }
+
+                    // If they are not the same, overwrite the old file with the new one
+                    if (!files_equal(in_file, out_file)) {
+                        log<NUClear::DEBUG>(fmt::format("Updating file {}", out_file.string()));
+                        fs::create_directories(out_file.parent_path());
+                        fs::copy_file(in_file, out_file, fs::copy_options::overwrite_existing);
+                    }
+                }
+            }
+
+            /***************
+             * PERMISSIONS *
+             ***************/
+            log<NUClear::INFO>("Applying permissions changes from configuration");
+            for (const auto& c : config["permissions"].config) {
+
+                fs::path path = c.first.as<std::string>();
+                log<NUClear::TRACE>(fmt::format("Checking file permissions for {}", path.string()));
+
+                if (fs::is_regular_file(path)) {
+                    // Apply permissions
+                    fs::perms old_perms = fs::status(path).permissions();
+                    fs::perms new_perms = static_cast<fs::perms>(c.second.as<int>());
+
+                    if (old_perms != new_perms) {
+                        log<NUClear::INFO>(fmt::format("Changing permissions on {} from {} to {}",
+                                                       path.string(),
+                                                       permissions_to_string(old_perms),
+                                                       permissions_to_string(new_perms)));
+                        fs::permissions(path, new_perms);
+                    }
+                }
+                else {
+                    log<NUClear::ERROR>(fmt::format("{} is not a regular file", path.string()));
+                }
+            }
+
+            /************
+             * SYMLINKS *
+             ************/
+            log<NUClear::INFO>("Ensuring relevant symlinks exist");
+            for (const auto& l : config["links"].config) {
+                std::string target = l.first.as<std::string>();
+                std::string link   = l.second.as<std::string>();
+                log<NUClear::TRACE>(fmt::format("Checking link {} -> {}", link, target));
+                if (fs::exists(target)) {
+                    log<NUClear::WARN>(fmt::format("Link target '{}' doesn't exist. Skipping", target));
```

Comment:
What is this branch for?

Reply:
Letting you know that you are trying to make a like to file that doesn't exist

Reply:
ok